### DISCUSSION TITLE
Is it possible to access the connection for the underlying dialect?

### DISCUSSION THREAD

**Discussion by shanecav84** [sequelize/sequelize#15038] [Help & Questions] [answered]
I'm using Sequelize for MSSQL and I'd like to add event subscriptions to tedious's connection. Specifically I'd like to subscriber to the `eventMessage` event so I can output the multiple errors that MSSQL sometimes returns. Sequelize only outputs the last error.

**Comment by ephys** [comment, marked as answer]
You can listen to the `afterConnect` event on your sequelize instance, it will be called every time a new connection is created

```
const sequelize = new Sequelize(/* config */);

sequelize.afterConnect(async (connection, config) => {
  // you should be able to add listeners to the connection object here
  // "connection" is a Connection instance from the "tedious" lib
});
```

> I can output the multiple errors that MSSQL sometimes returns. Sequelize only outputs the last error.

I've noticed this as well. This is something we desperately need to fix.

**Reply by shanecav84** [reply]
Thanks, that worked well.
